deepchecks/tabular/checks/data_integrity/is_single_value.py

```diff
@@ -59,29 +59,32 @@ def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResul
 
         df = select_from_dataframe(df, self.columns, self.ignore_columns)
 
-        is_single_unique_value = (df.nunique(dropna=False) == 1)
+        num_unique_per_col = df.nunique(dropna=False)
+        is_single_unique_value = (num_unique_per_col == 1)
 
         if is_single_unique_value.any():
             # get names of columns with one unique value
             # pylint: disable=unsubscriptable-object
             cols_with_single = is_single_unique_value[is_single_unique_value].index.to_list()
-            value = list(cols_with_single)
             uniques = df.loc[:, cols_with_single].head(1)
             uniques.index = ['Single unique value']
             display = ['The following columns have only one unique value', uniques]
         else:
-            value = None
             display = None
 
-        return CheckResult(value, header='Single Value in Column', display=display)
+        return CheckResult(num_unique_per_col.to_dict(), header='Single Value in Column', display=display)
 
     def add_condition_not_single_value(self):
         """Add condition - not single value."""
         name = 'Does not contain only a single value'
 
         def condition(result):
-            if result:
-                return ConditionResult(ConditionCategory.FAIL, f'Found columns with a single value: {result}')
-            return ConditionResult(ConditionCategory.PASS)
+            single_value_cols = [k for k, v in result.items() if v == 1]
+            if single_value_cols:
+                details = f'Found {len(single_value_cols)} columns with a single value out of {len(result)} ' \
+                          f'columns: {single_value_cols}'
```

Comment:
No need for total number of columns here in my opinion. Can be just "found X columns with single value: [...]"

Reply:
I agree with Matan